docs/typescript.md

```diff
@@ -2,89 +2,70 @@
+In Gasket, you can now write your `gasket.js` configuration file as `gasket.ts` using TypeScript. This allows for full type-checking and validation directly within your TypeScript environment.
 
-The `gasket.config.js` file cannot be written in TypeScript, but you can configure TypeScript to check JS files or use a `@ts-check` comment along with JSDoc type annotations. Due to JSDoc type checking limitations, you may have to separate the config declaration and your export:
+The `@gasket/core` package supplies a `GasketConfigFile` type that validates the contents of your Gasket config file. When you import all your plugins into `gasket.ts`, TypeScript can fully validate the configuration.
 
-```javascript
-//@ts-check
-///<reference types="@gasket/preset-nextjs"/>
-
-/** @type {import('@gasket/engine').GasketConfigFile} */
-const config = {
-  plugins: {
-    presets: ['@gasket/nextjs']
-  },
-  compression: true,
-  http: 8080,
-  intl: {
-    defaultLocale: 'en-GB'
-  }
-};
-
-module.exports = config;
-```
-
-## Validating lifecycle scripts
-
-The `@gasket/engine` package supplies a `Hook` type that can be used to validate lifecycle scripts and plugin hooks. It takes a string type parameter for the name of the lifecycle.
+Here's an example:
 
 ```typescript
-import { Hook } from '@gasket/engine';
-import '@gasket/plugin-intl';
-
-const intlLocaleHandler: Hook<'intlLocale'> = (gasket, locale, { req, res }) => {
-  // return 3; - does not pass validation
-  return getLocaleFromHost(req.headers.host);
-}
+// gasket.ts
+import type { GasketConfigDefinition } from '@gasket/core';
+import { makeGasket } from '@gasket/core';
+import pluginA from '@gasket/plugin-a';
+import pluginB from '@gasket/plugin-b';
+
+export default makeGasket({
+  plugins: [
+    pluginA,
+    pluginB
+  ]
+  // additional config
+} as GasketConfigDefinition);
 ```
 
-Gasket does not currently allow you to author `/lifecycles/*` files in TypeScript. To add type checking to these scripts you must either write them in TypeScript and make sure they get compiled to JavaScript or use JSDoc comments to enable checking. Because of JSDoc limitations with generics syntax, writing in JavaScript requires a workaround using a separate `@typedef` declaration.
+## Validating Lifecycle Hooks
 
-```javascript
-// @ts-check
-///<reference types="@gasket/plugin-intl"/>
+The `@gasket/core` package supplies a `HookHandler` type that can be used to validate lifecycle hooks in plugins. It takes a string type parameter for the name of the lifecycle, ensuring that your hooks conform to the expected signature.
 
-/** 
- * @typedef {import('@gasket/engine').Hook<'intlLocale'>} IntlLocaleHandler
- */
+```typescript
+import type { HookHandler } from '@gasket/core';
 
-/** @type {IntlLocaleHandler} */
-const handler = (gasket, currentLocale, { req, res }) => {
-  // return 3; - does not pass validation
+const intlLocaleHandler: HookHandler<'initLocale'> = (gasket, locale, { req, res }) => {
```

Comment:
Is the `///<reference types="@gasket/plugin-intl"/>` still necessary here?

Reply:
Maybe? I was testing with defining hook handlers and I didn't see an issues 🤷‍♂️